pandas/tests/io/json/test_pandas.py

```diff
@@ -1202,6 +1202,22 @@ def test_data_frame_size_after_to_json(self):
 
         assert size_before == size_after
 
+    def test_from_json_to_json_table_dtypes(self):
+        # GH21345
+        expected = pd.DataFrame({'a': [1, 2], 'b': [3., 4.], 'c': ['5', '6']})
+        dfjson = expected.to_json(orient='table')
+        result = pd.read_json(dfjson, orient='table')
+        assert_frame_equal(result, expected)
+
+    @pytest.mark.xfail(raises=ValueError)
```

Comment:
Use the `pytest.raises` context manager in the body of the test function rather than xfailing